model/host/host.go

```diff
@@ -380,6 +380,17 @@ func (h *Host) MarkAsProvisioned() error {
 			},
 		},
 	)
+	if err == nil {
+		h.Status = evergreen.HostRunning
+		h.Provisioned = true
+		h.ProvisionTime = time.Now()
+		return nil
+	}
+	if err == mgo.ErrNotFound {
```

Comment:
I think `mgo.ErrNotFound` means that nothing matched the update query parameters:
```
		bson.M{
			IdKey: h.Id,
			StatusKey: bson.M{
				"$nin": evergreen.DownHostStatus,
			},
		},
```
This either means there's no host with that ID (less likely) or that the host is not in the status we expected (more likely). In that case I think you _should_ return an error, because it means that the caller of `MarkAsProvisioned` should not keep doing work, since the host is down.

Reply:
Okay fixed!